compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1568,6 +1600,7 @@ impl<'tcx> TyCtxt<'tcx> {
         if *r == kind { r } else { self.mk_region(kind) }
     }
 
+    // Avoid this in favour of more specific `mk_*` methods, where possible.
```

Comment:
Should we just get rid of this?

Reply:
No. First, it's still used by all the other `mk_*` functions. Second, there are a couple of places remaining that are `Ty`-generic and still use it, e.g. [here](https://github.com/rust-lang/rust/blob/11d96b59307b1702fffe871bfc2d0145d070881e/compiler/rustc_middle/src/ty/structural_impls.rs#L681).